herddb-core/src/main/java/herddb/core/DBManager.java

```diff
@@ -907,6 +924,38 @@ public void dumpTableSpace(String tableSpace, String dumpId, Pdu message, Channe
         }
     }
 
+    public DataConsistencyStatementResult createTableCheksum(TableConsistencyCheckStatement tableIntegrityCheckStatement) {
+        TableSpaceManager manager = tablesSpaces.get(tableIntegrityCheckStatement.getTableSpace());
+        String table = tableIntegrityCheckStatement.getTable();
+        try {
+            manager.createAndWriteTableCheksum(manager, tableIntegrityCheckStatement.getTableSpace(), table);
+        } catch (IOException | DataScannerException ex) {
+            LOGGER.log(Level.SEVERE, null, ex);
```

Comment:
What about returning a DataConsistencyStatementResult that tells that we have an error ?
 return new DataConsistencyStatementResult(TransactionContext.NOTRANSACTION_ID, ex);